eureka/S6_planet_spectra/s6_spectra.py

```diff
@@ -0,0 +1,281 @@
+import numpy as np
+import pandas as pd
+import os, glob
+import time as time_pkg
+from ..lib import manageevent as me
+from ..lib import readECF as rd
+from ..lib import util, logedit
+from ..lib import sort_nicely as sn
+
+from . import plots_s6 as plots
+
+#FINDME: Keep reload statements for easy testing
+from importlib import reload
+
+class MetaClass:
+    '''A class to hold Eureka! metadata.
+    '''
+
+    def __init__(self):
+        return
+
+def plot_spectra(eventlabel, s5_meta=None):
+    '''Gathers together diferent wavelength fits and makes transmission/emission spectra.
+
+    Parameters
+    ----------
+    eventlabel: str
+        The unique identifier for these data.
+    s5_meta:    MetaClass
+        The metadata object from Eureka!'s S5 step (if running S5 and S6 sequentially).
+
+    Returns
+    -------
+    meta:   MetaClass
+        The metadata object with attributes added by S6.
+
+    Notes
+    -------
+    History:
+
+    - Feb 14, 2022 Taylor Bell
+        Original version
+    '''
+    print("\nStarting Stage 6: Light Curve Fitting\n")
+
+    # Initialize a new metadata object
+    meta = MetaClass()
+    meta.eventlabel = eventlabel
+
+    # Load Eureka! control file and store values in Event object
+    ecffile = 'S6_' + eventlabel + '.ecf'
+    ecf = rd.read_ecf(ecffile)
+    rd.store_ecf(meta, ecf)
+
+    # load savefile
+    if s5_meta == None:
+        s5_meta = read_s5_meta(meta)
+
+    meta = load_general_s5_meta_info(meta, s5_meta)
+
+    if (not meta.s5_allapers) or (not meta.allapers):
+        # The user indicated in the ecf that they only want to consider one aperture
+        # in which case the code will consider only the one which made s4_meta.
+        # Alternatively, S4 was run without allapers, so S6's allapers will only conside that one
+        meta.spec_hw_range = [meta.spec_hw,]
+        meta.bg_hw_range = [meta.bg_hw,]
+
+    if meta.testing_S6:
+        # Only fit a single channel while testing
+        chanrng = [0]
```

Comment:
This variable doesn't appear to be used, but may to keep this if statement for later.

Reply:
S5 was indeed not using chanrng, but it now is